src/crypto_wrappers/openssl/gen_cert.c

```diff
@@ -140,10 +142,16 @@ crypto_wrapper_err_t openssl_gen_cert(crypto_wrapper_ctx_t *ctx, rats_tls_cert_a
 
 	X509_set_version(cert, 3);
 	ASN1_INTEGER_set(X509_get_serialNumber(cert), CERT_SERIAL_NUMBER);
-	/* WORKAROUND: allow 1 hour delay for the systems behind current clock */
-	X509_gmtime_adj(X509_get_notBefore(cert), -3600);
-	/* 1 year */
-	X509_gmtime_adj(X509_get_notAfter(cert), (long)3600 * 24 * 365 * 1);
+	if (!cert_nonce_is_used) {
+		/* WORKAROUND: allow 1 hour delay for the systems behind current clock */
+		X509_gmtime_adj(X509_get_notBefore(cert), -3600);
+		/* 1 year */
+		X509_gmtime_adj(X509_get_notAfter(cert), (long)3600 * 24 * 365 * 1);
+	} else {
+		/* WORKAROUND: with nonce mechanism, the validity of cert can be fixed within a larger range. */
+		X509_gmtime_adj(X509_get_notBefore(cert), (long)-3600 * 24 * 365 * 30);
+		X509_gmtime_adj(X509_get_notAfter(cert), (long)3600 * 24 * 365 * 30);
```

Comment:
Can you uncomment this code https://github.com/inclavare-containers/rats-tls/blob/master/src/tls_wrappers/openssl/un_negotiate.c#L296 and try to dump the generated cert with your this this patch enabling using_cert_noce = 1?

The command is: openssl x509 -in /tmp/cert.der -inform der -text -noout

Reply:
Hi，thanks for your advice and comment and I have some confusions:
>about "uncomment this code....#L296"
1. The “uncomment this code” refers to my commit, or the link you posted?
2. I think it may means to  replace the "#if 0" with "using_cert_nonce=0"? The code in that "#if 0" contex looks like to retrieve the cert by cache, which  is valid when ra-tls don't use nonce, is it?
> about "try to dump..." 
3. I think it is to check the generated certificate whether the validity is fixed after this patch when enabling using_cert_noce = 1, is it?